### ISSUE TITLE
[a11y]: dropdown web component is focusable with VoiceOver when disabled

### ISSUE THREAD

**Issue by davidmenendez** [carbon-design-system/carbon#21982]
### Package

@carbon/web-components

### Browser

Chrome

### Operating System

MacOS

### Package version

v2.52.0

### Automated testing tool and ruleset

accessibility checker

### Assistive technology

voiceover

### Description

Reference issue from ai-chat https://github.com/carbon-design-system/carbon/issues/21987

During an accessibility audit of carbon for ai chat an issue was discovered where using voiceover that the disabled `cds-dropdown` component was still focusable. Unfortunately it doesn't look like you can use screen recording and VO at the same time, so I can't show the interaction here, but I've verified the issue is happening on the latest version of the dropdown web component. This creates an issue for keyboard users who end up with the disabled element in their tab path.

### WCAG 2.1 Violation

2.4.3

### Reproduction/example

https://web-components.carbondesignsystem.com/?path=/story/components-dropdown--default

### Steps to reproduce

* visit the dropdown web component [story](https://web-components.carbondesignsystem.com/?path=/story/components-dropdown--default&args=disabled:!true) and enable the `disabled` prop
* turn on voiceover
* click somewhere in the content area and press the tab button. it should apply focus to the dropdown

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by davidmenendez** [comment]
you can assign this to me 👍
